lib/package-manager/test.js

```diff
@@ -66,44 +66,72 @@ function test(packageManager, context, next) {
 
     /* Run `npm/yarn test`, or `/path/to/customTest.js` if the customTest option
      was passed */
-    const args = context.options.customTest
-      ? [context.options.customTest]
-      : [packageManagerBin, 'test'];
-
-    const proc = spawn(bin, args, options);
-    const finish = timeout(context, proc, next, 'Test');
-
-    proc.stdout.on('data', (data) => {
-      context.testOutput.append(data);
-      if (context.module.stripAnsi) {
-        data = stripAnsi(data.toString());
-        data = data.replace(/\r/g, '\n');
+    const scripts = context.options.customTest
+      ? [[context.options.customTest]]
+      : (context.module.scripts ? context.module.scripts : ['test']).map(
+          (x) => [packageManagerBin, 'run', x]
+        );
```

Comment:
Nit: wouldn't it be better to read if it's written as:

```js
let scripts;
if (context.options.customTest) {
  scripts = [[context.options.customTest]];
} else if (context.module.scripts) {
  scripts = context.module.scripts
    .map((script) => [packageManagerBin, 'run', script]);
} else {
  scripts = [packageManagerBin, 'run', 'test'];
}
```

Reply:
Fixed. FWIW `prettier` formats it slightly differently:
```diff
diff --git a/lib/package-manager/test.js b/lib/package-manager/test.js
index ba89636..39ae5b6 100644
--- a/lib/package-manager/test.js
+++ b/lib/package-manager/test.js
@@ -66,11 +66,18 @@ function test(packageManager, context, next) {

     /* Run `npm/yarn test`, or `/path/to/customTest.js` if the customTest option
      was passed */
-    const scripts = context.options.customTest
-      ? [[context.options.customTest]]
-      : (context.module.scripts ? context.module.scripts : ['test']).map(
-          (x) => [packageManagerBin, 'run', x]
-        );
+    let scripts;
+    if (context.options.customTest) {
+      scripts = [[context.options.customTest]];
+    } else if (context.module.scripts) {
+      scripts = context.module.scripts.map((script) => [
+        packageManagerBin,
+        'run',
+        script
+      ]);
+    } else {
+      scripts = [[packageManagerBin, 'test']];
+    }
     context.emit(
       'data',
       'silly',
```